File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/DestroyEntitiesPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record DestroyEntitiesPacket(int[] entityIds) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeByte(entityIds.length);
        writeIntArray(os, entityIds);
    }

    @Override
    public byte getId() {
        return 0x13;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/DestroyEntitiesPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class DestroyEntitiesPacket(val entityIds: IntArray) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeByte(entityIds.size)
        writeIntArray(os, entityIds)
    }

    override fun getId(): Byte {
        return 0x13
    }
}
